tests/eval_files/trio100.py

```diff
@@ -35,17 +44,35 @@ async def function_name():
         async for _ in receive_channel:
             pass
 
+    # fix missed alarm when function is defined inside the with scope
+    with trio.move_on_after(10):  # error: 4,"trio", "move_on_after"
+
+        async def foo():
+            await trio.sleep(1)
+
+    with trio.move_on_after(10):  # error: 4,"trio", "move_on_after"
+        if ...:
+
+            async def foo():
+                if ...:
+                    await trio.sleep(1)
+
+    async with random_ignored_library.fail_after(10):
+        pass
+
 
 async def function_name2():
-    with (
+    # now throws the error at the with statement
+    with (  # error: 4, "trio", "fail_after"
         open("") as _,
-        trio.fail_after(10),  # error: 8, "trio", "fail_after"
+        trio.fail_after(10),
```

Comment:
Hmm, I prefer the more precise (old) locations - can we get that back, maybe [via `WithItem`](https://libcst.readthedocs.io/en/latest/nodes.html#libcst.WithItem)?

Reply:
Fixed! ^^